### DISCUSSION TITLE
Pesan gambar dengan URL query string

### DISCUSSION THREAD

**Discussion by rioastamal** [kirimwa-id/forum#53] [Tanya Jawab]
Ini adalah pertanyaan yang saya dapat dari email.

---

Halo mas Rio

Saya ingin bertanya lagi,
saya menggunakan service firebase untuk menyimpan gambar yang user upload lalu saya ambil url nya untuk di kirim ke v1/messages berikut url dari firebase :

```
https://firebasestorage.googleapis.com/v0/b/example.appspot.com/o/whatsapp_broadcaster%2Fcris%20-%20Copy.jpg?alt=media&token=some-token-xyz
```

tapi kembalian dari api tersebut menyatakan :
"Failed to send message. Error: Invalid image format, allowed format are JPEG, PNG and GIF."

apakah ada solusi untuk masalah tersebut?
terimakasih.

**Comment by rioastamal** [comment]
Bug confirmed, akan diperbaiki pada versi mendatang.

**Comment by rioastamal** [comment]
Setelah saya lakukan test ulang ternyata tidak ada bug seperti yang saya sangka sebelumnya. Kemungkinan URL tersebut tidak mengembalikan mime type yang diinginkan yaitu antara `['image/jpeg', 'image/png', 'image/gif']`. API KirimWA.id menggunakan `HTTP HEAD` request untuk melakukan pengecekan content.

**Reply by rioastamal** [reply]
Ini terjadi karena server tidak menerima HTTP `HEAD` request.

**Comment by arhaviolet** [comment]
Sepertinya kasus saya seperti ini. Kirim dokumen dan teks lancar tp pada saat gambar selalu gagal

**Reply by rioastamal** [reply]
Sertakan code atau CLI command dari `curl` dengan mensensor informasi yang sensitif agar lebih mudah dipahami.

Pastikan juga gambar yang diserver tujuan menerima HTTP HEAD request. Contoh

```sh
curl --head 'https://rioastamal.net/portfolio/img/rioastamal.jpg?build=202001140325'
```

```
HTTP/2 200 
date: Tue, 21 Jan 2025 05:23:35 GMT
content-type: image/jpeg
content-length: 16199
cache-control: public, max-age=14400, must-revalidate
cache-status: "Netlify Edge"; fwd=miss
etag: "578d22f415b31acbd0c7cb78f890862e-ssl"
x-nf-request-id: 01JHY3HEZKDX4W0BTEPT2QJ91C
cf-cache-status: REVALIDATED
accept-ranges: bytes
report-to: {"endpoints":[{"url":"https:\/\/a.nel.cloudflare.com\/report\/v4?s=QUja7CW6ne1T4o0WRyelCmclBu9w0TH1nejs9mxMj%2F7xLha1JNe20tN6n6%2BfqWATvWwjzuTySrF3Uy1SbueFs8S9HBkr4bTj8PvgC5Qt8nEYADzy0xmLHbZKbBBHNwjEXA%3D%3D"}],"group":"cf-nel","max_age":604800}
nel: {"success_fraction":0,"report_to":"cf-nel","max_age":604800}
server: cloudflare
cf-ray: 9054df846f5f2c5d-FRA
alt-svc: h3=":443"; ma=86400
server-timing: cfL4;desc="?proto=TCP&rtt=5469&min_rtt=5379&rtt_var=1214&sent=5&recv=9&lost=0&retrans=0&sent_bytes=3396&recv_bytes=815&delivery_rate=524162&cwnd=252&unsent_bytes=0&cid=00c1b7efdf6a24b6&ts=36&x=0"
```

**Reply by arhaviolet** [reply]
Hasilnya seperti ini mas 

Curl --head 'https://www.appsheet.com/template/gettablefileurl?appName=InspeksiMesinASP-912669164&tableName=Kontrol%20Produk&fileName=Kontrol%20Produk_Images%2F01-20-2025%2016-15-00.Foto%20Purity%20FBT%20LOX.091811.jpg'

`HTTP/1.1 200 OK
Content-Length: 2603995
server: istio-envoy
x-cloud-trace-context: c1a6b5a60b7b7c51bd8670f98844742a/3745339172333447285;o=0
strict-transport-security: max-age=63072000; includeSubDomains; preload
x-frame-options: DENY
content-security-policy: object-src 'none'; base-uri 'none'; script-src 'nonce-lFQYzJZ4J0MF6lrpQ92r9g==' 'strict-dynamic' 'report-sample' 'unsafe-eval' 'unsafe-inline' 'self' https://www.gstatic.com/ https://www.google-analytics.com/gtm/ https://www.googletagmanager.com/gtag/ https://storage.googleapis.com/appsheet-staging-frontend-assets https://storage.googleapis.com/appsheet-prod-frontend-assets https://www.youtube.com; worker-src  'self'; report-uri https://csp.withgoogle.com/csp/appsheet/2;
content-security-policy: default-src 'none'; media-src 'self'
content-security-policy: sandbox
accept-range: bytes
x-content-type-options: nosniff
content-disposition: inline; filename*=utf-8''Kontrol%20Produk_Images%2F01-20-2025%2016-15-00.Foto%20Purity%20FBT%20LOX.091811.jpg
cross-origin-opener-policy: same-origin
x-content-security-policy: sandbox
x-envoy-upstream-service-time: 3464
via: 1.1 google
Date: Tue, 21 Jan 2025 05:38:26 GMT
Content-Type: image/jpeg; charset=UTF-8
Age: 360
Cache-Control: public,max-age=3600,s-maxage=3600
X-client_region: US
X-client_region_subdivision: USNJ
Alt-Svc: h3=":443"; ma=2592000,h3-29=":443"; ma=2592000`

**Reply by rioastamal** [reply]
Saya tidak bisa mencobanya karena sekarang sudah 404 not found.

```sh
curl --head 'https://www.appsheet.com/template/gettablefileurl?appName=InspeksiMesinASP-912669164&tableName=Kontrol%20Produk&fileName=Kontrol%20Produk_Images%2F01-20-2025%2016-15-00.Foto%20Purity%20FBT%20LOX.091811.jpg'
```

```
HTTP/2 404 
date: Tue, 21 Jan 2025 23:02:22 GMT
server: istio-envoy
cache-control: no-store
x-cloud-trace-context: ba21bd40a48a93f41db061607255aff0/7033147136774916017;o=0
x-cloud-trace-context: ba21bd40a48a93f41db061607255aff0/7033147136774916017;o=0
strict-transport-security: max-age=63072000; includeSubDomains; preload
x-frame-options: DENY
content-security-policy: object-src 'none'; base-uri 'none'; script-src 'nonce-9c5Yv7XRLdRpMLqLjGgzSQ==' 'strict-dynamic' 'report-sample' 'unsafe-eval' 'unsafe-inline' 'self' https://www.gstatic.com/ https://www.google-analytics.com/gtm/ https://www.googletagmanager.com/gtag/ https://storage.googleapis.com/appsheet-staging-frontend-assets https://storage.googleapis.com/appsheet-prod-frontend-assets https://www.youtube.com; worker-src  'self'; require-trusted-types-for 'script'; report-uri https://csp.withgoogle.com/csp/appsheet/2;
content-security-policy-report-only: require-trusted-types-for 'script'; report-uri https://csp.withgoogle.com/csp/appsheet/3;
x-envoy-upstream-service-time: 2
via: 1.1 google
vary: Accept-Encoding
x-client_region: DE
x-client_region_subdivision: 
alt-svc: h3=":443"; ma=2592000,h3-29=":443"; ma=2592000
```
